File: AnalyticAlways.AngularTest.Domain/Store.cs
namespace AnalyticAlways.AngularTest.Domain
{
    public class Store
    {
        protected Store()
        { }

        public Store(string name, string city)
        {
            Name = name;
            City = city;
        }

        public int Id { get; private set; }

        public string Name { get; private set; }

        public string City { get; private set; }
    }
}

File: angular-test2/src/app/stores/shared/store.ts
export class Store {

    public id: number;
    public name: string;
    public city: string;

    public static parse(json: any): Store {
        return new Store(
            json.id,
            json.name,
            json.city);
    }

    public static parseCollection(json: any): Store[] {
        return json.map(item => {
            return Store.parse(item);
        });
    }

    constructor(
        id: number,
        name: string,
        city: string
    ) {
        this.id = id;
        this.name = name;
        this.city = city;
    }
}
